web/src/lib/utils.ts

```diff
@@ -194,3 +195,13 @@ export const findLocale = (code: string | undefined) => {
     name: language?.name,
   };
 };
+
+export const asyncTimeout = (ms: number) => {
+  return new Promise((resolve) => {
+    setTimeout(resolve, ms);
+  });
+};
+
+export const resolvePromise = <T>(promise: Promise<T>): void => {
+  Promise.resolve(promise).catch(async (error) => await handleError(error));
+};
```

Comment:
Why is `Promise.resolve` needed when only promises are accepted as input? Also I don't think the name `resolvePromise` properly communicates that the error will get handled.

Reply:
All this does it automatically notify the user when an error happens. Maybe something like this would be better `notifyOnError(() => loadFavorites())`

Reply:
updated